pkg/utils/csr_helpers.go

```diff
@@ -178,3 +182,28 @@ func UnionCSRApprover(approvers ...agent.CSRApproveFunc) agent.CSRApproveFunc {
 		return true
 	}
 }
+
+// IsCSRSupported checks whether the cluster supports v1 or v1beta1 csr api.
+func IsCSRSupported(nativeClient kubernetes.Interface) (bool, bool, error) {
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(nativeClient.Discovery()))
```

Comment:
this will add additional rbac requirement for addon controller

Reply:
the `system:discovery` clusterrole will be bound to all the authenticated users. so no additional permission is required.

```
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:authenticated
```

Reply:
ah ok